test/e2e/framework/util.go

```diff
@@ -3825,19 +3825,39 @@ func RestartKubeProxy(host string) error {
 
 func RestartKubelet(host string) error {
 	// TODO: Make it work for all providers and distros.
-	if !ProviderIs("gce", "aws") {
-		return fmt.Errorf("unsupported provider: %s", TestContext.Provider)
+	supportedProviders := []string{"gce", "aws", "vsphere"}
+	if !ProviderIs(supportedProviders...) {
+		return fmt.Errorf("unsupported provider: %s, supported providers are: %v", TestContext.Provider, supportedProviders)
 	}
 	if ProviderIs("gce") && !NodeOSDistroIs("debian", "gci") {
```

Comment:
How about some minor simplification:
```
var cmd string
if ProviderIs("gce") {
      if !NodeOSDistroIs("debian", "gci") {
            return fmt.Errorf("unsupported provider: %s, supported providers...
      }
      cmd = "sudo /etc/init.d/kubelet restart"
} else if ProviderIs("vsphere") { 
      ...
} else { ...
}
```

Reply:
I see your reason here to avoid redundant checks for NodeOSDistroIs

however, `cmd = "sudo /etc/init.d/kubelet restart" ` is only valid for `debian` and not for `gci`

So this simplification might not work correctly.

Reply:
ah, ok then never mind...